transcendence-app/src/game/infrastructure/db/postgres/game.postgres.repository.ts

```diff
@@ -71,4 +76,34 @@ export class GamePostgresRepository
     });
     return gamesData ? gamesData.map((game) => new this.ctor(game)) : null;
   }
+
+  async getPaginatedUserGamesWithUsers(
+    username: string,
+    paginationDto: Required<PaginationWithSearchQueryDto>,
+  ): Promise<GameWithUsers[] | null> {
+    const { limit, offset, sort, search } = paginationDto;
+    const orderBy =
+      sort === BooleanString.True ? gameKeys.CREATED_AT : gameKeys.ID;
+    const gamesData = await makeQuery<GameWithUsersData>(this.pool, {
+      text: `SELECT g.*,
+      ${renameColumnsWithPrefix('p1', Object.values(userKeys), 'playerOne')},
+      ${renameColumnsWithPrefix('p2', Object.values(userKeys), 'playerTwo')}
+      FROM ${this.table} g
+      JOIN ${table.USERS} AS p1 ON g.${gameKeys.PLAYERONEUSERNAME} = p1.${
+        userKeys.USERNAME
+      }
+      JOIN ${table.USERS} AS p2 ON g.${gameKeys.PLAYERTWOUSERNAME} = p2.${
+        userKeys.USERNAME
+      }
+      WHERE (g.${gameKeys.PLAYERONEUSERNAME} = $4
+        AND g.${gameKeys.PLAYERTWOUSERNAME} ILIKE $1)
+      OR (g.${gameKeys.PLAYERTWOUSERNAME} = $4
+        AND g.${gameKeys.PLAYERONEUSERNAME} ILIKE $1)
```

Comment:
why do we want to make this case insensitive?

Reply:
I don't mind, we can change it to `LIKE $1` if you want